wormhole_chain/x/wormhole/client/cli/tx_wasmd.go

```diff
@@ -0,0 +1,153 @@
+			return types.MsgStoreCode{}, err
+		}
+	} else if !ioutils.IsGzip(wasm) {
+		return types.MsgStoreCode{}, fmt.Errorf("invalid input file. Use wasm binary or gzip")
+	}
+	return types.MsgStoreCode{
+		Signer:       sender.String(),
+		WASMByteCode: wasm,
+		Vaa:          vaa,
+	}, nil
+}
+
+// StoreCodeCmd will upload code to be reused.
+func CmdStoreCode() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "store [wasm file] [vaa-hex]",
+		Short:   "Upload a wasm binary with vaa, or just compute the hash for vaa if [vaa-hex] is omitted",
+		Aliases: []string{"upload", "st", "s"},
+		Args:    cobra.RangeArgs(1, 2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+			hash_only := len(args) == 1
+			vaaBz := []byte{}
+			if !hash_only {
+				vaaBz, err = hex.DecodeString(args[1])
+				if err != nil {
+					return err
+				}
+			}
+
+			msg, err := parseStoreCodeArgs(args[0], clientCtx.GetFromAddress(), vaaBz)
+			if err != nil {
+				return err
+			}
+
+			if !hash_only {
+				if err = msg.ValidateBasic(); err != nil {
+					return err
+				}
+				return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), &msg)
+			} else {
+				var hashWasm [32]byte
+				keccak := sha3.NewLegacyKeccak256()
+				keccak.Write(msg.WASMByteCode)
+				keccak.Sum(hashWasm[:0])
+				fmt.Println(hex.EncodeToString(hashWasm[:]))
+				return nil
+			}
+		},
+	}
+
+	flags.AddTxFlagsToCmd(cmd)
+	return cmd
+}
+
+func CmdInstantiateContract() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "instantiate [label] [code_id_int64] [json_encoded_init_args] [vaa-hex]",
+		Short: "Register a guardian public key with a wormhole chain address, or just compute the hash for vaa if vaa is omitted",
+		Args:  cobra.RangeArgs(3, 4),
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+			hash_only := len(args) == 3
+
+			labelStr := args[0]
+
+			codeId, err := cast.ToUint64E(args[1])
+			if err != nil {
+				return err
+			}
+
+			initMsg := args[2]
+
+			vaaBz := []byte{}
+			if !hash_only {
+				vaaBz, err = hex.DecodeString(args[3])
+				if err != nil {
+					return err
+				}
+			}
+
+			msg := types.MsgInstantiateContract{
+				Signer: clientCtx.GetFromAddress().String(),
+				CodeID: codeId,
+				Label:  labelStr,
+				Msg:    []byte(initMsg),
+				Vaa:    vaaBz,
+			}
+			if !hash_only {
+				if err := msg.ValidateBasic(); err != nil {
+					return err
+				}
+				return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), &msg)
+			} else {
+				hash_base := make([]byte, 8)
+				binary.BigEndian.PutUint64(hash_base, msg.CodeID)
+				hash_base = append(hash_base, []byte(msg.Label)...)
+				hash_base = append(hash_base, []byte(msg.Msg)...)
+				var hash [32]byte
+				keccak := sha3.NewLegacyKeccak256()
+				keccak.Write(hash_base)
+				keccak.Sum(hash[:0])
```

Comment:
`hash.Hash` implements `io.Write` so there's no need to build an array first:

```suggestion
				var hash [32]byte
				keccak := sha3.NewLegacyKeccak256()
				binary.Write(keccak, binary.BigEndian, msg.CodeID)
				keccak.Write([]byte(msg.Label))
				keccak.Write([]byte(msg.Msg))
				keccak.Sum(hash[:0])
```